dm/syncer/job.go

```diff
@@ -228,6 +233,24 @@ func newConflictJob(workerCount int) *job {
 	}
 }
 
+// newCompactJob is only used for metrics.
+func newCompactJob(targetTable *filter.Table) *job {
+	return &job{
+		tp:          compact,
+		targetTable: targetTable,
+	}
+}
+
+// newDummyJob is only used in tests.
+func newDummyJob(tp opType, targetTable *filter.Table, ddls ...string) *job {
```

Comment:
Nit: We can move this function to _test.go since it is a package private function.